### PR TITLE
New Features

### PR DESCRIPTION
**Pull request by ruancomelli**
Bank Selector Window
Login Updates
Custom Components
New Assets

## Summary by Sourcery

This pull request introduces a new Bank Selector window, updates the login window with custom components, and adds new assets for these components. It also includes enhancements to the layout and background image handling in the login window.

- **New Features**:
    - Introduced a new Bank Selector window for selecting bank sites.
    - Added custom components including Custom_Button, Custom_PasswordField, and Custom_TextField.
    - Implemented new assets for the custom components and login background.
- **Enhancements**:
    - Updated the login window to use new custom components and improved layout.
    - Refined the background image loading and rendering in Custom_Panel.

### CODE REVIEW

**Comment by ruancomelli** [comment]
This is a benchmark review for experiment `improve_review_guide`.
Run ID: `improve_review_guide/benchmark_2024-08-06T17-40-52_v1-21-0-70-g56484f776`.

This pull request was cloned from `https://github.com/sourcery-ai-experiments/PED/pull/3`. _(Note: the URL is not a link to avoid triggering a notification on the original pull request.)_
<details>
<summary>Experiment configuration</summary>

```yaml
review_config:
  # User configuration for the review
  # - benchmark - use the user config from the benchmark reviews
  # - <value> - use the value directly
  user_review_config:
    enable_ai_review: true
    enable_rule_comments: false

    enable_complexity_comments: false
    enable_security_comments: false
    enable_tests_comments: false
    enable_comment_suggestions: false
    enable_functionality_review: false

    enable_pull_request_summary: false
    enable_review_guide: true

    enable_approvals: false

  ai_review_config:
    # The model responses to use for the experiment
    # - benchmark - use the model responses from the benchmark reviews
    # - llm - call the language model to generate responses
    model_responses:
      comments_model: benchmark
      comment_area_model: benchmark
      comment_validation_model: benchmark
      comment_suggestion_model: benchmark
      complexity_model: benchmark
      docstrings_model: benchmark
      functionality_model: benchmark
      security_model: benchmark
      tests_model: benchmark
      pull_request_summary_model: benchmark
      review_guide_model: llm

# The pull request dataset to run the experiment on
pull_request_dataset:
  # open-source PRs
- https://github.com/sourcery-ai-experiments/gpt-crawler/pull/1
- https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/1
- https://github.com/sourcery-ai-experiments/graphviz2drawio/pull/2
- https://github.com/sourcery-ai-experiments/iptux/pull/5
- https://github.com/sourcery-ai-experiments/iptux/pull/6
- https://github.com/sourcery-ai-experiments/kalima/pull/3
- https://github.com/sourcery-ai-experiments/PED/pull/3
- https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/1
- https://github.com/sourcery-ai-experiments/pydantic_tfl_api/pull/2
- https://github.com/sourcery-ai-experiments/Trenova/pull/11
  # Sourcery PRs
- https://github.com/sourcery-ai/core/pull/4055
- https://github.com/sourcery-ai/core/pull/4263
- https://github.com/sourcery-ai/core/pull/4352
- https://github.com/sourcery-ai/core/pull/4364
- https://github.com/sourcery-ai/core/pull/4366
- https://github.com/sourcery-ai/core/pull/4390
- https://github.com/sourcery-ai/core/pull/4402
- https://github.com/sourcery-ai/core/pull/4445
- https://github.com/sourcery-ai/core/pull/4462
- https://github.com/sourcery-ai/core/pull/4463

review_comment_labels: []

# Benchmark reviews generated by running
#   python -m scripts.experiment benchmark <experiment_name>
benchmark_reviews: []

```

</details>

**Comment by SourceryAI** [comment]
## Reviewer's Guide by Sourcery

This pull request introduces a new Bank Selector window, updates the login window with custom components, and adds new assets for these components. It also includes enhancements to the layout and background image handling in the login window.

### File-Level Changes

| Files | Changes |
| ----- | ------- |
| `PED/src/gui/w_Login.java`<br/>`PED/src/presets/Custom_Panel.java` | Updated login window to use new custom components and improved layout. |
| `PED/src/gui/W_BankSelector.java`<br/>`PED/src/PED.java` | Introduced new Bank Selector window and updated main application to use it. |
| `PED/src/presets/Custom_Button.java`<br/>`PED/src/presets/Custom_PasswordField.java`<br/>`PED/src/presets/Custom_TextField.java` | Added new custom components for buttons, password fields, and text fields. |

---

<details>
<summary>Tips</summary>

- Trigger a new Sourcery review by commenting `@sourcery-ai review` on the pull request.
- Continue your discussion with Sourcery by replying directly to review comments.
- You can change your review settings at any time by accessing your [dashboard](https://app.sourcery.ai):
    - Enable or disable the Sourcery-generated pull request summary or reviewer's guide;
    - Change the review language;
- You can always [contact us](mailto:support@sourcery.ai) if you have any questions or feedback.

</details>

File: PED/src/gui/w_Login.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public void run() {
         
         
         pnl_StatusBar.showPanel(appSettings);
-        pnl_Login.showPanel("BG_Login.png");
+        pnl_Login.showPanel("BG1.png");
```
Comment (SourceryAI) [inline PED/src/gui/w_Login.java:37]: **suggestion:** Consider using a configuration setting or constant for the image name.

Hardcoding the image name can lead to maintainability issues. Consider using a configuration setting or a constant to define the image name, making it easier to change in the future if needed.

File: PED/src/presets/Custom_Panel.java [hunk 2]
```diff
@@ -27,17 +26,11 @@ public Custom_Panel() {
...
-        System.out.println("Esta es la dirección de la imagen  : " + url + "/n"
-                + "El file existe o no : " + backgroundImage.getProperty("key", this));
-        
-        
-        tf_username.setOpaque(false);
-        tf_username.setBackground(new java.awt.Color(0, 0, 0, 0));
-        tf_password.setOpaque(false);
-        tf_password.setBackground(new java.awt.Color(0, 0, 0, 0));
+        //System.out.println("Esta es la dirección de la imagen  : " + url + "/n"
+        //        + "El file existe o no : " + backgroundImage.getProperty("key", this));
```
Comment (SourceryAI) [inline PED/src/presets/Custom_Panel.java:33]: **suggestion:** Remove commented-out debug print statements.

Leaving commented-out code can lead to confusion and clutter. If these debug print statements are no longer needed, consider removing them entirely.

File: PED/src/presets/Custom_Panel.java [hunk 3]
```diff
@@ -27,17 +26,11 @@ public Custom_Panel() {
     public void showPanel(String imageName){
         initComponents();
         
-        // Cargar la imagen de fondo desde el directorio assets
+        // Carga la imagen de fondo desde la carpeta assets
```
Comment (SourceryAI) [inline PED/src/presets/Custom_Panel.java:29]: **suggestion (testing):** Test the changes in paintComponent method for drawing the background image.

The changes in the paintComponent method to draw the background image should be tested to ensure they work as expected. Verify that the image scaling logic is correct and doesn't introduce any visual artifacts.

File: PED/src/presets/Custom_PasswordField.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        
+        // Cargar la imagen de fondo desde el directorio assets
+        backgroundImage = new ImageIcon("src/assets/TextField.png").getImage();
+        setOpaque(false);
+        repaint();
+    
+    }
+    
+    // Convierte el password de char[] a un String
+    public String getPWSD(){
```
Comment (SourceryAI) [inline PED/src/presets/Custom_PasswordField.java:33]: **🚨 issue (security):** Review the security implications of the getPWSD method.

The getPWSD method converts the password from a char array to a String, which can be a security risk as Strings are immutable and can be stored in memory longer than necessary. Consider alternative approaches to handle passwords securely.

**Review by SourceryAI — COMMENTED** [review]
Hey @ruancomelli - I've reviewed your changes and found some issues that need to be addressed.

**Blocking issues**:
- Review the security implications of the getPWSD method. ([link](https://github.com/sourcery-ai-experiments/PED/pull/7/files#diff-8b05e050cb613dda852f6528582baff593f9ae9b90d176f3443230a51565044dR33))

**Overall Comments**:
- Consider using a more flexible approach for loading resources instead of hard-coded paths. This will improve portability and maintainability.
- Remove commented-out code and debug print statements to keep the codebase clean and maintainable.
- Refactor the layout management to be more dynamic instead of hard-coded values. This will make the GUI more adaptable to different screen sizes and resolutions.
- Consider using a more flexible approach for loading resources instead of hard-coded paths. This will improve portability and maintainability.
- Remove commented-out code and debug print statements to keep the codebase clean and maintainable.
- Refactor the layout management to be more dynamic instead of hard-coded values. This will make the GUI more adaptable to different screen sizes and resolutions.

<details>
<summary>Here's what I looked at during the review</summary>

- 🟡 **General issues**: 2 issues found
- 🟢 **Documentation**: all looks good
</details>

***

[LangSmith trace](https://smith.langchain.com/o/7d8518ad-6dd3-5c7b-96d2-757769296335/projects/p/88560225-5072-48bc-967c-dc54da53f265/r/c6b176b3-6381-4481-bc7b-21e2b8ca5392?poll=true)

<sub>
Help me be more useful! Please click 👍 or 👎 on each comment to tell me if it was helpful.
</sub>

**Pull request closed without merging**